tools/crd-bumper/README.md

```diff
@@ -0,0 +1,117 @@
+# CRD Version Bumper
+
+Bump the CRD version, adding conversion webhooks and tests.
+
+See Kubebuilder's [Tutorial: Multi-Version API](https://book.kubebuilder.io/multiversion-tutorial/tutorial) for a description of the mechanism. For more detail read the Kubernetes document [Versions in CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/).
+
+## Hub and Spokes
+
+This tool implements the hub-and-spoke model of CRD versioning. See Kubebuilder's [Hubs, spokes, and other wheel metaphors](https://book.kubebuilder.io/multiversion-tutorial/conversion-concepts) for a description of the model.
+
+The new CRD version will be the new hub, and the previous hub will become a new spoke. (Note: in the Kubebuilder book example, the old version is the hub and never changes, and all new versions are spokes.)
+
+# Using CRD Bumper
+
+## Create Environment
+
+```console
+$ python3 -m venv venv
+$ . venv/bin/activate
+(venv) $ pip install -r requirements.txt
+```
+
+
+## Run the Tool
+
+Clone a fresh copy of the repository that contains the CRDs and controllers, checked out to the default branch (master or main). The tool expects a repository that is compatible with kubebuilder and will use the `./PROJECT` file that is maintained by kubebuilder.
+
+The following example will create a new API version "v1beta2" for the lustre-fs-operator repository, where "v1beta1" is the existing hub and "v1alpha1" is the most recent existing spoke. It begins by creating a new branch off "master" named "api-v1beta2", where it will do all of its work.
+
+```console
+git clone git@github.com:NearNodeFlash/lustre-fs-operator.git
+cd lustre-fs-operator
+crd-bumper.py --most-recent-spoke v1alpha1 --prev-ver v1beta1 --new-ver v1beta2 all
```

Comment:
Won't all of these commands happen within the `(venv)`? Also, the instructions above assume your in this directory, so do we need to add the cloned folders to `.gitignore`?

Reply:
Now the tool does the clone for you, and clones it under `workingspace/lustre-fs-operator` (in this case). That dir is already covered by .gitignore.